bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/split/coordinator/KuduSourceSplitCoordinator.java

```diff
@@ -60,21 +56,20 @@ public KuduSourceSplitCoordinator(Context<KuduSourceSplit, EmptyState> context,
 
   @Override
   public void start() {
-    List<KuduSourceSplit> splitList;
-    try (KuduFactory kuduFactory = KuduFactory.initReaderFactory(jobConf)) {
-      KuduClient client = kuduFactory.getClient();
-      AbstractKuduSplitConstructor splitConstructor = KuduSplitFactory.getSplitConstructor(jobConf, client);
-      splitList = splitConstructor.construct(client);
-    } catch (IOException e) {
-      throw new BitSailException(KuduErrorCode.SPLIT_ERROR, "Failed to create splits.");
-    }
+    List<KuduSourceSplit> splitList = SimpleDivideSplitConstructor.getInstance(jobConf).construct();
 
     int readerNum = context.totalParallelism();
     LOG.info("Found {} readers and {} splits.", readerNum, splitList.size());
     if (readerNum > splitList.size()) {
       LOG.error("Reader number {} is larger than split number {}.", readerNum, splitList.size());
     }
 
+    if (splitList.isEmpty()) {
+      String errorMsg = "empty kudu split";
+      LOG.error(errorMsg);
+      throw new BitSailException(KuduErrorCode.SPLIT_ERROR, errorMsg);
+    }
+
```

Comment:
It is recommend directly read an empty split instead of throw exception.
Because it is common that upstream data source is empty.

Reply:
The empty split will case the KuduReader dead loop